thin-replica-server/include/thin-replica-server/thin_replica_impl.hpp

```diff
@@ -347,10 +347,12 @@ class ThinReplicaImpl {
             update_aggregator_counter = 0;
           }
         }
-        config_->subscriber_list.removeBuffer(live_updates);
-        live_updates->removeAllUpdates();
-        metrics_.subscriber_list_size.Get().Set(config_->subscriber_list.Size());
-        metrics_.updateAggregator();
+        if (!is_event_group_transition) {
```

Comment:
We can even put the whole `try-catch` block under this condition. Good catch! 👍

Reply:
It is [this](https://github.com/vmware/concord-bft/blob/c48238d8501f1309dffd9db007762ff0b429788f/thin-replica-server/include/thin-replica-server/thin_replica_impl.hpp#L428) call to removeBuffer that results in the crash. 
I was thinking we add a `try-catch` block inside the `removeBuffer` method, and remove the assert, WDYT?

Reply:
I was thinking of throwing a runtime exception. But I can't think of a situation where this error will be seen other than a coding error, so I agree, assert is the right way to go.